src/libs/Navigation/AppNavigator/MainDrawerNavigator.js

```diff
@@ -78,7 +79,11 @@ const MainDrawerNavigator = (props) => {
             isMainScreen
         />
     );
-};
+}, (prevProps, nextProps) => {
+    const initialPrevParams = getInitialReportScreenParams(prevProps.reports, !Permissions.canUseDefaultRooms(prevProps.betas), prevProps.policies);
+    const initialNextParams = getInitialReportScreenParams(nextProps.reports, !Permissions.canUseDefaultRooms(nextProps.betas), nextProps.policies);
+    return initialPrevParams.reportID === initialNextParams.reportID;
```

Comment:
I think since `getInitialReportScreenParams()` can in some cases return an empty string (not sure what those are) it would be safer to check that there is both a value and it is the same as previous.

So something like:

```js
return !_.isEmpty(initialNextParams.reportID) && (initialPrevParams.reportID === initialNextParams.reportID);
```

Reply:
Not a blocker, but I think it is maybe slightly less efficient to do the `memo()` than something like a class component with `shouldComponentUpdate()` since that lifecycle method can refer to an instance property like `this.initialParams.reportID` that is already calculated on the previous `render()`.

I bet in practice the impact here would be very small, but I think with large numbers of reports it's possible that the sort and filter performed here are still happening on each update:

https://github.com/vladamx/App/blob/71e021721eeeedced60122c9265e6eb9230939d9/src/libs/ReportUtils.js#L232-L240

Reply:
> I think since `getInitialReportScreenParams()` can in some cases return an empty string (not sure what those are) it would be safer to check that there is both a value and it is the same as previous.
> 
> So something like:
> 
> ```js
> return !_.isEmpty(initialNextParams.reportID) && (initialPrevParams.reportID === initialNextParams.reportID);
> ```

Wouldn't it be better if we do  
`return initialPrevParams.reportID === initialNextParams.reportID || _.isEmpty(initialNextParams.reportID);`
In that way if initialNextParams is empty will still evaluate the whole expression to true so we can skip the render, otherwise the result would be the same(false) as if there was no check to isEmpty.

Reply:
>Wouldn't it be better if we do
`return initialPrevParams.reportID === initialNextParams.reportID || _.isEmpty(initialNextParams.reportID);`
In that way if `initialNextParams` is empty will still evaluate the whole expression to true so we can skip the render, otherwise the result would be the same(false) as if there was no check to isEmpty.

I don't know if this alternative is equivalent, but maybe it doesn't matter. I was thinking if the `initialPrevParams.reportID` is an empty string (for some reason) and the `initialNextParams.reportID` is an empty string then we would say the props are equal and the component would not render. 

In reality, that might be fine since once we do have a value the initial `reportID` will be set. Practically speaking, as long as the fallback loader still shows up when there is no `reportID` we should be good.

>didn't want to rewrite function to class component,which is deprecated API, just for that

AFAIK there are no plans to remove class components from React. So we use them freely here 😄

Reply:
Okay, i will then use class based shouldComponentUpdate and leave the comparison initialPrevParams.reportID === initialNextParams.reportID as is.

Reply:
@marcaaron Updated

Reply:
Thanks!